docs/adr/adr-009-public-api.md

```diff
@@ -204,12 +204,17 @@ SyncHead(ctx context.Context) (*header.ExtendedHeader, error)
 
   type NodeModule interface {
     // Type returns the node type.
-    Type() node.Type
+    Type(context.Context) node.Type
     // Version returns information about the current binary build.
-    Version() string
+    Version(context.Context) string
```

Comment:
* Do we really need to return the build version? Usually the version is accessed through the CLI
* We want to introduce API versioning, so if we want to show both the build and API versions, we should introduce a struct containing those already so that we break our API less.

Reply:
Ah, I got confused. You return a structure, but this is an ADR, so we should update the ADR.

Reply:
I'm actually okay to 
1. omit the build version from the Version endpoint
2. combine Type and the api info into one single struct called `Info` that contains fields `Type` and `APIVersion`

Reply:
Implemented, but API version implementation will be later PR